infection_monkey/network/network_scanner.py

```diff
@@ -36,8 +36,26 @@ def initialize(self):
         self._ranges = [NetworkRange.get_range_obj(address_str=x) for x in WormConfiguration.subnet_scan_list]
         if WormConfiguration.local_network_scan:
             self._ranges += get_interfaces_ranges()
+        self._ranges += self._get_inaccessible_subnets_ips()
         LOG.info("Base local networks to scan are: %r", self._ranges)
 
+    def _get_inaccessible_subnets_ips(self):
+        subnets_to_scan = []
+        for subnet_group in WormConfiguration.inaccessible_subnet_groups:
+            for subnet_str in subnet_group:
+                if NetworkScanner._is_any_ip_in_subnet([unicode(x) for x in self._ip_addresses], subnet_str):
+                    # If machine has IPs from 2 different subnets in the same group, there's no point checking the other
+                    # subnet.
```

Comment:
More conceptual, but if we have a machine in both subnets as a possible pinpoint, don't we want to alert on that?

Reply:
Interesting point. I fear we'll get a lot of false positives (in cases we'll have cross segmentation issues). Also unsure how strong this is. Feels awkward to add a section of machines which may be a gateway between the subnets. What do you think?

Reply:
Fixed